session/vertexai/service_test.go

```diff
@@ -0,0 +1,1282 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vertexai
+
+import (
+	"context"
+	"maps"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/rpcreplay"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+	"google.golang.org/api/option"
+	"google.golang.org/genai"
+	"google.golang.org/grpc"
+
+	"google.golang.org/adk/model"
+	"google.golang.org/adk/session"
+)
+
+const (
+	ProjectID = "adk-go-test"
+	Location  = "us-central1"
+	EngineId  = "5576569044451983360"
+	EngineId2 = "8602987994044956672"
+	UserID    = "test-user"
+)
+
+func Test_vertexaiService_Create(t *testing.T) {
+	tests := []struct {
+		name       string
+		setup      func(t *testing.T, name string) (session.Service, map[string]string)
+		req        *session.CreateRequest
+		want       session.Session
+		wantErr    bool
+		errMessage string
+	}{
+		{
+			name:  "full key",
+			setup: emptyService,
+			req: &session.CreateRequest{
+				AppName:   EngineId,
+				UserID:    "testUserID",
+				SessionID: "testSessionID",
+				State: map[string]any{
+					"k": 5,
+				},
+			},
+			wantErr:    true,
+			errMessage: "user-provided Session id is not supported for VertexAISessionService: \"testSessionID\"",
+		},
+		{
+			name:  "generated session id",
+			setup: emptyService,
+			req: &session.CreateRequest{
+				AppName: EngineId,
+				UserID:  "testUserID",
+				State: map[string]any{
+					// TODO had to parse to float64, sending int was modified by vertex or by vertex client, int should work
+					"k": float64(5),
+				},
+			},
+		},
+		{
+			name:  "when already exists, it fails",
```

Comment:
how this test case is different from `full key`? Both provide custom `session_id` and fails due to this.